sdc11073/mdib/devicemdib.py

```diff
@@ -495,56 +452,77 @@ def _process_transaction(self, setDeterminationTime):
-                                        mgr.contextStateUpdates, mgr.operationalStateUpdates):
-                        update =  update_dict.get(descriptorContainer.handle)
-                        if update is not None:
-                            oldstate, newstate = update
-                            if newstate is not None: # this is a create or update operation
-                                corresponding_state = newstate
-                                corresponding_state.descriptorContainer = descriptorContainer
-                                corresponding_state.updateDescriptorVersion()
-                                self._logger.info(
-                                    'mdibUpdateTransaction: Descriptor update move state update to descriptor update:{}',
-                                    newstate)
-                                del update_dict[descriptorContainer.handle]
-                                multikey_instance = self.contextStates if newstate.isContextState else self.states
-                                if oldstate:
-                                    multikey_instance.removeObjectNoLock(oldstate)
-                                multikey_instance.addObjectNoLock(newstate)
-                            break
-
-                    if corresponding_state is None:
-                        corresponding_state = self.states.descriptorHandle.getOne(descriptorContainer.handle,
-                                                                                  allowNone=True)
-                        if corresponding_state is not None:
-                            corresponding_state.descriptorContainer = descriptorContainer
-                            corresponding_state.incrementState()
-                            corresponding_state.updateDescriptorVersion()
-                    if corresponding_state is not None:
-                        updated_states.append(corresponding_state)
+                    # add state to updated_states list and to corresponding notifications input
+                    # => the state is always sent twice, a) in the description modification report and b)
+                    # in the specific state update notification.
+                    if descriptorContainer.isAlertDescriptor:
+                        update_dict = mgr.alertStateUpdates
+                    elif descriptorContainer.isComponentDescriptor:
+                        update_dict = mgr.componentStateUpdates
+                    elif descriptorContainer.isContextDescriptor:
+                        update_dict = mgr.contextStateUpdates
+                    elif descriptorContainer.isMetricDescriptor \
+                            and not descriptorContainer.isRealtimeSampleArrayMetricDescriptor:
+                        update_dict = mgr.metricStateUpdates
+                    elif descriptorContainer.isOperationalDescriptor:
+                        update_dict = mgr.operationalStateUpdates
+                    else:

```

Comment:
This should still be an elif, should be above `elif descriptorContainer.isMetricDescriptor`, and should have an else case for errors, if you accidentally missed a case, meaning
```python
...
elif descriptorContainer.isRealtimeSampleArrayMetricDescriptor:
    update_dict = mgr.rtSampleStateUpdates
elif descriptorContainer.isMetricDescriptor:
    update_dict = mgr.metricStateUpdates
...
else:
   raise SomeKindOfException("Because I don't know what this was")
```